packages/caravan-fees/src/btcTransactionComponents.ts

```diff
@@ -0,0 +1,331 @@
+    this._address = params.address;
+    this._malleable = !params.locked;
+
+    if (!this._malleable && this._amountSats.isEqualTo(0)) {
+      throw new Error("Locked outputs must have an amount specified.");
+    }
+  }
+
+  /** Get the recipient address */
+  get address(): string {
+    return this._address;
+  }
+
+  /** Check if the output is malleable (can be modified) */
+  get isMalleable(): boolean {
+    return this._malleable;
+  }
+
+  /**
+   * Set a new amount for the output
+   * @param amountSats - New amount in satoshis(as a string)
+   * @throws Error if trying to modify a non-malleable output
+   */
+  setAmount(amountSats: Satoshis): void {
+    if (!this._malleable) {
+      throw new Error("Cannot modify non-malleable output");
+    }
+    this._amountSats = new BigNumber(amountSats);
+  }
+
+  /**
+   * Add amount to the output
+   * @param amountSats - Amount to add in satoshis (as a string)
+   * @throws Error if trying to modify a non-malleable output
+   */
+  addAmount(amountSats: Satoshis): void {
+    if (!this._malleable) {
+      throw new Error("Cannot modify non-malleable output");
+    }
+    this._amountSats = this._amountSats.plus(new BigNumber(amountSats));
+  }
+
+  /**
+   * Subtract amount from the output
+   * @param amountSats - Amount to subtract in satoshis (as a string)
+   * @throws Error if trying to modify a non-malleable output or if subtracting more than the current amount
+   */
+  subtractAmount(amountSats: Satoshis): void {
+    if (!this._malleable) {
+      throw new Error("Cannot modify non-malleable output");
+    }
+    const subtractAmount = new BigNumber(amountSats);
+    if (subtractAmount.isGreaterThan(this._amountSats)) {
+      throw new Error("Cannot subtract more than the current amount");
+    }
+    this._amountSats = this._amountSats.minus(subtractAmount);
+  }
+
+  /**
+   * Locks the output, preventing further modifications to its amount.
+   *
+   * This method sets the malleability of the output to false. Once called,
+   * the output amount cannot be changed, and attempts to do so will throw an error.
+   * This operation is irreversible - once locked, an output cannot be unlocked.
+   *
+   * Typical use cases include:
+   * - Finalizing a transaction before signing
+   * - Ensuring that certain outputs (like recipient amounts) are not accidentally modified
+   *
+   * For EXTERNAL outputs, an amount must be specified before locking. This is to prevent
+   * locking an output with an undefined amount, which could lead to invalid transactions.
+   *
+   * @throws {Error} If the output is already locked (non-malleable)
+   * @throws {Error} If trying to lock an EXTERNAL output with an undefined amount
+   */
+  lock(): void {
+    if (!this.isMalleable) {
+      throw new Error("Output is already locked and cannot be modified.");
+    }
+
+    this._malleable = false;
+  }
+
+  /**
+   * Checks if the output is valid according to basic Bitcoin transaction rules.
```

Comment:
I think these code comments are slightly out of date now?

Reply:
Yes .. corrected the docstrings [now](https://github.com/caravan-bitcoin/caravan/pull/114/commits/372f9197eb1a321131f3a18e1639f6d14f321e56#diff-db5fc022ab08b92fb2a606342674432f918748c762c9fcb907bc43d3531d512bR310)